java/algorithm/src/main/java/org.apache.arrow.algorithm.sort/FixedWidthOutOfPlaceVectorSorter.java

```diff
@@ -41,39 +39,41 @@ public V sort(V srcVector, VectorValueComparator<V> comparator) {
     int valueWidth = comparator.getValueWidth();
 
     // create output vector
-    V dstVector = comparator.newVector(srcVector.getAllocator());
-    dstVector.allocateNew(srcVector.getValueCount());
+    V dstVector = createOutputVector(srcVector.getAllocator(), srcVector);
 
     // buffers referenced in the sort
     ArrowBuf srcValueBuffer = srcVector.getDataBuffer();
     ArrowBuf dstValidityBuffer = dstVector.getValidityBuffer();
     ArrowBuf dstValueBuffer = dstVector.getDataBuffer();
 
     // sort value indices
-    int[] sortedIndices = IntStream.range(0, srcVector.getValueCount()).toArray();
-    indexSorter.sort(sortedIndices, comparator);
+    try (IndexSorter<V> indexSorter = new IndexSorter<>()) {
+      indexSorter.sort(srcVector, comparator);
+      IntVector sortedIndices = indexSorter.getSortedIndices();
 
-    // copy sorted values to the output vector
-    int dstIndex = 0;
-    for (int srcIndex : sortedIndices) {
-      if (srcVector.isNull(srcIndex)) {
-        BitVectorHelper.setValidityBit(dstValidityBuffer, dstIndex, 0);
-      } else {
-        BitVectorHelper.setValidityBit(dstValidityBuffer, dstIndex, 1);
-        PlatformDependent.copyMemory(
-                srcValueBuffer.memoryAddress() + srcIndex * valueWidth,
-                dstValueBuffer.memoryAddress() + dstIndex * valueWidth,
-                valueWidth);
+      // copy sorted values to the output vector
+      for (int dstIndex = 0; dstIndex < sortedIndices.getValueCount(); dstIndex++) {
+        int srcIndex = sortedIndices.get(dstIndex);
+        if (srcVector.isNull(srcIndex)) {
+          BitVectorHelper.setValidityBit(dstValidityBuffer, dstIndex, 0);
+        } else {
+          BitVectorHelper.setValidityBit(dstValidityBuffer, dstIndex, 1);
+          PlatformDependent.copyMemory(
+                  srcValueBuffer.memoryAddress() + srcIndex * valueWidth,
+                  dstValueBuffer.memoryAddress() + dstIndex * valueWidth,
+                  valueWidth);
+        }
       }
-      dstIndex += 1;
     }
 
     dstVector.setValueCount(srcVector.getValueCount());
     return dstVector;
   }
 
   @Override
-  public boolean isInPlace() {
-    return false;
+  public V createOutputVector(BufferAllocator allocator, V inputVector) {
```

Comment:
probably should not be public..

Reply:
Thanks for the comments. This method is public because it is overriding a function in an interface. All methods in an interface must be public.

I can think of two solutions for this problem, but I have not determined which one is better:

1. Make OutOfPlaceVectorSorter an abstract class instead of an interface. 
2. Remove the method altogether.

Reply:
Sure. It is removed in the new commit.